src/pybind/mgr/prometheus/module.py

```diff
@@ -1015,6 +993,31 @@ def add_fixed_name_metrics(self):
 
         self.metrics.update(new_metrics)
 
+    @profile_method(True)
+    def get_collect_time_metrics(self):
+        if 'prometheus_collect_duration_seconds_sum' not in self.metrics:
+            self.metrics['prometheus_collect_duration_seconds_sum'] = MetricCounter(
+                'prometheus_collect_duration_seconds_sum',
+                'The sum of seconds took to collect all metrics of this exporter',
+                ('method',),
+            )
+        if 'prometheus_collect_duration_seconds_count' not in self.metrics:
+            self.metrics['prometheus_collect_duration_seconds_count'] = MetricCounter(
+                'prometheus_collect_duration_seconds_count',
+                'The amount of metrics gathered for this exporter',
+                ('method',),
+            )
+
+        # Collect all timing data and make it available as metric, excluding
+        # this `collect` method because it has not finished at this point and
+        # hence there's no `_execution_duration` attribute to be found.
```

Comment:
This could use some additional commentary and how the `_execution_duration` attribute gets there in the first place. Some future maintainer would probably appreciate being pointed to the `profile_method` decorator here.